spec/integration/apply_error_spec.rb

```diff
@@ -52,7 +52,9 @@
     let(:password) { conn_info('winrm')[:password] }
     let(:tflags) { %w[--no-ssl --no-ssl-verify] }
 
-    it 'prints a helpful error if Puppet is not present' do
+    # This test currently won't pass because all windows integration tests
+    # are run on the same VM, which has puppet-agent installed.
+    xit 'prints a helpful error if Puppet is not present' do
```

Comment:
Hmm, I think there are a few possible solutions here:
* The right way to do this is to set up the Windows environment so that it's connecting to a remote Puppet agent (probably a VM but maybe containers would work now?) just like on *nix. That's something we've wanted to do for a long time, but is a lot of work and out of scope for this PR
* We could run this test in the `integration` test job before installing the agent (so have a `bundle exec rake ci:windows:agentless` then run `scripts/ci.ps1` then `bundle exec rake ci:windows:integration`)
* We could run this in addition to the `unit` test job (so `bundle exec rake tests:unit` then `bundle exec rake ci:windows:agentless`)
* We could have a separate job for agentless Windows integration tests.
I'd lean towards a separate job, but totally up to you and @beechtom

Reply:
I think I lean towards having a separate job for now. I agree that long-term it'd be nice to use containers, but it's definitely not a priority.